packages/connection-storage/src/connection-storage.spec.ts

```diff
@@ -230,6 +232,55 @@ describe('ConnectionStorage', function () {
       );
       expect(expectedConnectionInfo).to.deep.equal(connectionInfo);
     });
+
+    context(`max allowed connections ${maxAllowedConnections}`, function () {
+      const createNumberOfConnections = (num: number) => {
+        const connectionInfos = Array.from({ length: num }, (v, i) =>
+          getConnectionInfo({
+            lastUsed: new Date(1690876213077 - (i + 1) * 1000), // Difference of 1 sec
+          })
+        );
+        connectionInfos.forEach((connectionInfo) =>
+          writeFakeConnection(tmpDir, { connectionInfo })
+        );
+
+        return connectionInfos;
+      };
+
+      it('truncates recents to max allowed connections', async function () {
+        const connectionInfos = createNumberOfConnections(
+          maxAllowedConnections
```

Comment:
I would suggest to do the same things as earlier in the maximum number of queries:

```suggestion
          maxAllowedConnections + 1
```

To force surpassing the limit. What do you think?